### PR TITLE
add FastRNG to ax ns and more improvements 

### PR DESCRIPTION
**Pull request by DelinWorks**
## Describe your changes
#2056

### CODE REVIEW

File: core/math/FastRNG.h [hunk 1]
```diff
@@ -28,6 +25,10 @@
 #ifndef __FAST_RNG_H__
 #define __FAST_RNG_H__
 
+#include<math/MathBase.h>
```
Comment (halx99) [inline core/math/FastRNG.h]:
```suggestion
#include "math/MathBase.h"
```

**Comment by halx99** [comment]
also should add `include <stdint.h>`

File: core/math/FastRNG.h [hunk 2]
```diff
@@ -26,6 +26,8 @@
 #define __FAST_RNG_H__
 
 #include "math/MathBase.h"
+#include <iostream>
```
Comment (halx99) [inline core/math/FastRNG.h]: well, iostream not required

Comment (DelinWorks) [reply core/math/FastRNG.h]: for std::is_same

Comment (DelinWorks) [reply core/math/FastRNG.h]: I guess iostream replaced with type_traits

**Pull request merged**